microdata_tools/validation/steps/dataset_validator.py

```diff
@@ -274,7 +276,7 @@ def from_epoch_days_to_date(epoch_days: Union[int, None]) -> str:
             )
         )
 
-    def find_overlap(start_list, stop_list) -> Union[str, None]:
+    def find_overlap(start_list: list, stop_list: list) -> Union[str, None]:
```

Comment:
```suggestion
    def find_overlap(start_list: list, stop_list: list) -> str | None:
```

Reply:
Could do this more places too 👍🏻 The new way looks better. Not important tho so feel free to wait until another PR if it clutters up this one

Reply:
Ah! Of course! Well, once 3.9 is EOL then 🕐